### ISSUE TITLE
[1.16.4] Interacting with the GUI of a Board Computer crashes the game

### ISSUE THREAD

**Issue by qrss** [mcenderdragon/Futurepack-API#718]
Forge version: 35.1.13
Futurepack version: 21.0.2

When you have a (Advanced) Board Computer, hovering over the red X crashes the game.

If the Board Computer is part of a correctly formed spaceship, and there is a green checkmark instead of the X, then there is no crash.

Error when rendering screen: java.lang.ArrayStoreException

[debug.log](https://github.com/mcenderdragon/Futurepack-API/files/5745321/debug.log)
[latest.log](https://github.com/mcenderdragon/Futurepack-API/files/5745322/latest.log)
[FP-Latest.log](https://github.com/mcenderdragon/Futurepack-API/files/5745323/FP-Latest.log)

![board_computer_crash](https://user-images.githubusercontent.com/32312262/103180941-6b2a5800-489b-11eb-85fd-1a448df6d327.gif)

```
java.lang.ArrayStoreException: net.minecraft.util.IReorderingProcessor$$Lambda$4484/1357311524
	at com.google.common.collect.ImmutableList.copyIntoArray(ImmutableList.java:588) ~[guava-21.0.jar:?] {}
	at com.google.common.collect.ImmutableCollection.toArray(ImmutableCollection.java:201) ~[guava-21.0.jar:?] {}
	at futurepack.common.gui.inventory.GuiBoardComputer.renderButtons(GuiBoardComputer.java:396) ~[futurepack:1.16.4-21.0.2] {re:classloading}
	at futurepack.common.gui.inventory.GuiBoardComputer.func_230450_a_(GuiBoardComputer.java:271) ~[futurepack:1.16.4-21.0.2] {re:classloading}
	at net.minecraft.client.gui.screen.inventory.ContainerScreen.func_230430_a_(ContainerScreen.java:88) ~[?:?] {re:classloading,pl:runtimedistcleaner:A}
	at futurepack.common.gui.inventory.GuiBoardComputer.func_230430_a_(GuiBoardComputer.java:91) ~[futurepack:1.16.4-21.0.2] {re:classloading}
	at net.minecraftforge.client.ForgeHooksClient.drawScreen(ForgeHooksClient.java:290) ~[forge:?] {re:classloading}
	at net.minecraft.client.renderer.GameRenderer.func_195458_a(GameRenderer.java:491) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_195542_b(Minecraft.java:976) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:shutupexperimentalsettings.mixins.json:client.MixinMinecraft,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:607) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:shutupexperimentalsettings.mixins.json:client.MixinMinecraft,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:184) ~[1.16.4-forge-35.1.13.jar:?] {re:classloading,pl:runtimedistcleaner:A}
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_202] {}
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_202] {}
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_202] {}
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_202] {}
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider.lambda$launchService$0(FMLClientLaunchProvider.java:51) ~[forge-1.16.4-35.1.13.jar:35.1] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-8.0.6.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54) [modlauncher-8.0.6.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72) [modlauncher-8.0.6.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:82) [modlauncher-8.0.6.jar:?] {}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:66) [modlauncher-8.0.6.jar:?] {}
```

**Comment by mcenderdragon** [comment]
PLease try this version [21.0.3](https://www.mediafire.com/file/nqyc5vmwp6rnjvk/Futurepack-1.16.4-21.0.3.jar/file)

**Comment by qrss** [comment]
Good! Hovering over the red X no longer crashes the game.

However, I found a different crash in the GUI.

If you spam left click (it seems double-click is enough) on the empty slot where the Coordinates go, the game crashes.

The game crashed whilst container click
Error: java.lang.ClassCastException: net.minecraft.client.entity.player.ClientPlayerEntity cannot be cast to net.minecraft.entity.player.ServerPlayerEntity
Exit Code: -1

[crash-2020-12-30_19.11.13-client.txt](https://github.com/mcenderdragon/Futurepack-API/files/5755230/crash-2020-12-30_19.11.13-client.txt)
[debug.log](https://github.com/mcenderdragon/Futurepack-API/files/5755231/debug.log)
[FP-Latest.log](https://github.com/mcenderdragon/Futurepack-API/files/5755233/FP-Latest.log)
[latest.log](https://github.com/mcenderdragon/Futurepack-API/files/5755234/latest.log)

**Comment by mcenderdragon** [comment]
Should be fixed in [21.0.4 ](https://www.mediafire.com/file/mmmrw65ckdc34xn/Futurepack-1.16.4-21.0.4.jar/file)

**Comment by qrss** [comment]
Haven't had the Board Computer cause any crashes in 21.0.4, so it seems to be fixed!
